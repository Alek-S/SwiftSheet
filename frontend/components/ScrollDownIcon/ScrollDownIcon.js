import React from 'react';
import styled, { keyframes } from 'styled-components';

/** @function
 * @name ScrollDownIcon
 * animated mouse scroll icon
 * based on: https://codepen.io/shakdaniel/pen/JoKOQx
 * @returns {JSX}
 */
const ScrollDownIcon = () => (
	<ScrollDownAnimation>
		<p>Scroll Down</p>
	</ScrollDownAnimation>
);

const scrollAnimation = keyframes`
	0% {
		opacity: 1;
	}
	100% {
		opacity: 0;
		-webkit-transform: translateY(46px);
		transform: translateY(46px);
	}
`;

const ScrollDownAnimation = styled.div`
	width: 40px;
	height: 70px;
	margin-left: -20px;
	top: 50%;
	margin-top: -35px;
	box-shadow: inset 0 0 0 1px ${props => props.theme.color.red};
	background-color: rgba(255, 255, 255, 0.5);
	border-radius: 25px;

	&,
	&:before {
		position: absolute;
		left: 50%;
		top: calc(100vh - 150px);
		z-index: 10;
	}

	&:before {
		content: '';
		width: 8px;
		height: 8px;
		background: ${props => props.theme.color.red};
		margin-left: -4px;
		top: 8px;
		border-radius: 4px;
		-webkit-animation-duration: 1.5s;
		animation-duration: 1.5s;
		-webkit-animation-iteration-count: infinite;
		animation-iteration-count: infinite;
		-webkit-animation-name: scroll;
		animation-name: ${scrollAnimation};
	}

	p {
		position: relative;
		bottom: -80px;
		color: ${props => props.theme.color.red};
		font-family: ${props => props.theme.font.main};
		font-size: 1rem;
		font-weight: 400;
		text-align: center;
		line-height: 120%;
	}

	@media (max-width: 475px) {
		margin-top: -85px;
	}
`;

export default ScrollDownIcon;
