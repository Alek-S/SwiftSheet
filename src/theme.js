/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
