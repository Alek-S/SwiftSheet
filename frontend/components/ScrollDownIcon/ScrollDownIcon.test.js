import React from 'react';
import 'jest-styled-components';

import ScrollDownIcon from './ScrollDownIcon';
import theme from '../../theme';

describe('ScrollDownIcon', () => {
	test('matches the snapshot', () => {
		const tree = shallowWithTheme(<ScrollDownIcon />, theme);
		expect(toJson(tree)).toMatchSnapshot();
	});
});
