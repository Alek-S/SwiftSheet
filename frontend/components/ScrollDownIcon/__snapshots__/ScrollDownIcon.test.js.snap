// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollDownIcon matches the snapshot 1`] = `
<ScrollDownIcon__ScrollDownAnimation>
  <p>
    Scroll Down
  </p>
</ScrollDownIcon__ScrollDownAnimation>
`;
